tracing/tracers/otel/otel.go

```diff
@@ -0,0 +1,320 @@
+)
+
+type tracerBuilder struct {
+	exporter *otlptrace.Exporter
+	provider *sdk.TracerProvider
+	opt      options
+}
+
+type options struct {
+	customHeaders  map[string]string
+	host           string
+	port           int
+	useHTTP        bool
+	ServiceName    string
+	ServiceVersion string
+	Environment    string
+	customPath     string
+	insecure       bool
+
+	exportTimeout      time.Duration
+	batchTimeout       time.Duration
+	maxExportBatchSize int
+	maxSpanQueueSize   int
+	compressionMethod  string
+	tags               map[string]string
+}
+
+func InitTracer(ctx context.Context, opts []string) (trace.Tracer, error) {
+	opt, err := parseOptions(opts)
+	if err != nil {
+		return nil, err
+	}
+
+	builder := tracerBuilder{
+		opt: opt,
+	}
+
+	return builder.Build(ctx)
+}
+
+func WithBridgeTracer(ctx context.Context, opts []string) (trace.Tracer, ot.Tracer, error) {
+	opt, err := parseOptions(opts)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	builder := tracerBuilder{
+		opt: opt,
+	}
+
+	t, err := builder.Build(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	_, otTracer, wrapper := otelBridge.NewTracerPairWithContext(ctx, t)
+	otel.SetTracerProvider(wrapper)
+	return t, otTracer, nil
+}
+
+func parseOptions(opts []string) (options, error) {
+	config := options{}
+	globalTags := make(map[string]string)
+
+	for _, o := range opts {
+		key, val, _ := strings.Cut(o, "=")
+		switch key {
+		case "custom-headers":
+			config.customHeaders = make(map[string]string)
+			err := json.Unmarshal([]byte(val), &config.customHeaders)
+			if err != nil {
+				return options{}, fmt.Errorf("failed to parse %q as key-value custom headers: %w", val, err)
+			}
+		case "service-name":
+			if val != "" {
+				config.ServiceName = val
+			}
+		case "service-version":
+			if val != "" {
+				config.ServiceVersion = val
+			}
+		case "max-export-batch-size":
+			v, err := strconv.Atoi(val)
+			if err != nil {
+				return options{}, fmt.Errorf("failed to parse %q as int max-export-batch-size: %w", val, err)
+			}
+			config.maxExportBatchSize = v
+		case "max-span-queue-size":
+			v, err := strconv.Atoi(val)
+			if err != nil {
+				return options{}, fmt.Errorf("failed to parse %q as int max-export-batch-size: %w", val, err)
+			}
+			config.maxSpanQueueSize = v
+		case "batch-timeout":
+			v, err := time.ParseDuration(val)
+			if err != nil {
+				return options{}, fmt.Errorf("failed to parse %q as time.Duration min-period : %w", val, err)
+			}
+			config.batchTimeout = v
+		case "export-timeout":
+			v, err := time.ParseDuration(val)
+			if err != nil {
+				return options{}, fmt.Errorf("failed to parse %q as time.Duration min-period : %w", val, err)
+			}
+			config.exportTimeout = v
+		case "tag":
+			if val != "" {
+				tag, tagVal, found := strings.Cut(val, "=")
+				if !found {
+					return options{}, fmt.Errorf("missing value for tag %q", val)
+				}
+				globalTags[tag] = tagVal
+			}
+		case "otlp-exporter-endpoint":
```

Comment:
From testing in the cluster: maybe an URL?

Reply:
Yeah, I'll try to test with a local OTLP collector to make sure no parameter is missing. But one of the things we need to change is this parameter should not be split by port and address only because we need the URL path to route it to the proper OTLP endpoint in most collectors.